controllers/component_manager.go

```diff
@@ -39,13 +39,19 @@ func NewComponentManager(
 	cfgen := ytconfig.NewGenerator(resource, clusterDomain)
 	nodeCfgGen := &cfgen.NodeGenerator
 
+	var allComponents []components.Component
+
+	getAllComponents := func() []components.Component {
+		return allComponents
+	}
+
 	d := components.NewDiscovery(cfgen, ytsaurus)
 	m := components.NewMaster(cfgen, ytsaurus)
 	var hps []components.Component
 	for _, hpSpec := range ytsaurus.GetResource().Spec.HTTPProxies {
 		hps = append(hps, components.NewHTTPProxy(cfgen, ytsaurus, m, hpSpec))
 	}
-	yc := components.NewYtsaurusClient(cfgen, ytsaurus, hps[0])
+	yc := components.NewYtsaurusClient(cfgen, ytsaurus, hps[0], getAllComponents)
```

Comment:
Have you considered to have different virtual component like cypressPatcher or something, so we don't have those weird cyclic dependencies between components and yt client?